custom_components/unifiprotect/config_flow.py

```diff
@@ -36,79 +49,123 @@ class UnifiProtectFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
         )
 
-        unifiprotect = UpvServer(
-            session,
-            user_input[CONF_HOST],
-            user_input[CONF_PORT],
-            user_input[CONF_USERNAME],
-            user_input[CONF_PASSWORD],
+        if CONF_HOST in user_input:
+            host = user_input[CONF_HOST]
+            port = user_input.get(CONF_PORT, DEFAULT_PORT)
+        # reauth flow, pull host/port from existing settings
+        elif entry:
+            host = entry.data[CONF_HOST]
+            port = entry.data.get(CONF_PORT, DEFAULT_PORT)
+        else:
+            return None, {}
+
+        protect = UpvServer(
+            session=session,
+            host=host,
+            port=port,
+            username=user_input[CONF_USERNAME],
+            password=user_input[CONF_PASSWORD],
         )
 
+        errors = {}
+        nvr_data = None
         try:
-            server_info = await unifiprotect.server_information()
-            if server_info["server_version"] < MIN_REQUIRED_PROTECT_V:
+            nvr_data = await protect.server_information()
+            if nvr_data["server_version"] < MIN_REQUIRED_PROTECT_V:
                 _LOGGER.debug("UniFi Protect Version not supported")
                 errors["base"] = "protect_version"
-                return await self._show_setup_form(errors)
-
         except NotAuthorized as ex:
             _LOGGER.debug(ex)
             errors["base"] = "connection_error"
-            return await self._show_setup_form(errors)
         except NvrError as ex:
             _LOGGER.debug(ex)
             errors["base"] = "nvr_error"
-            return await self._show_setup_form(errors)
 
-        unique_id = server_info[SERVER_ID]
-        server_name = server_info[SERVER_NAME]
+        return nvr_data, errors
 
-        await self.async_set_unique_id(unique_id)
-        self._abort_if_unique_id_configured()
+    @callback
+    def _get_config_entry(self) -> Optional[config_entries.ConfigEntry]:
+        if self.unique_id is None:
+            return None
 
-        return self.async_create_entry(
-            title=server_name,
-            data={
-                CONF_ID: server_name,
-                CONF_HOST: user_input[CONF_HOST],
-                CONF_PORT: user_input[CONF_PORT],
-                CONF_USERNAME: user_input.get(CONF_USERNAME),
-                CONF_PASSWORD: user_input.get(CONF_PASSWORD),
-            },
-            options={
-                CONF_DISABLE_RTSP: False,
-                CONF_DOORBELL_TEXT: "",
-            },
+        for entry in self._async_current_entries(include_ignore=True):
+            if entry.unique_id == self.unique_id:
+                return entry
+
+        return None
+
+    async def async_step_reauth(self, user_input: Dict[str, Any] = None) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+
+        errors = {}
+        if user_input is not None:
+            entry = self._get_config_entry()
+            if not entry:
```

Comment:
```suggestion
            if not (entry := self._get_config_entry()):
```

Reply:
Decided not to us the walrus operator since it causes a multi-line if (with the other changes to get rid of `_get_config_entry`.